### PR TITLE
Fix scheduler cache panic when updating pod condition

### LINKED ISSUE #56730: Back-port scheduler cache panic fix to 1.8 and 1.7
There was a bug in scheduler which is fixed in 1.9 and later as a part of [this PR](https://github.com/kubernetes/kubernetes/pull/55039). We need to backport it to 1.8 and 1.7.

The sequence of events to trigger this issue is:
 - Pod fails predicates, pod condition was updated with the predicate error.
 - Pod passes predicates
 - Pod fails bind, pod condition was updated with the bind error.

The Conditions in assumedPod is the same ptr as in pod because [assumedPod is a shallow copy](https://github.com/kubernetes/kubernetes/blob/release-1.8/plugin/pkg/scheduler/scheduler.go#L320), not a DeepCopy of pod.
Changing the assumedPod condition causes a cache panic.

/kind bug
/sig scheduling
/assign

### PR DESCRIPTION
**Pull request by bsalamat**
**What this PR does / why we need it**:
This PR fixes an issue in the scheduler code which could cause a cache panic. More details are given in the referenced issue.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #56730

**Special notes for your reviewer**:

**Release note**:

```release-note
Fix scheduler cache panic when updating pod conditions.
```

/sig scheduling

### CODE REVIEW

**Comment by bsalamat** [comment]
@davidopp or @timothysc Could one of you guys add 1.7 milestone to this PR please?

File: plugin/pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -270,16 +270,21 @@ func (sched *Scheduler) scheduleOne() {
 
 	// Tell the cache to assume that a pod now is running on a given node, even though it hasn't been bound yet.
 	// This allows us to keep scheduling without waiting on binding to occur.
-	assumedPod := *pod
+	podCopy, err := api.Scheme.DeepCopy(pod)
```
Comment (timothysc) [inline plugin/pkg/scheduler/scheduler.go]: Could you add a comment to why you are doing this, it's totally non-obvious.

**Review by timothysc — CHANGES_REQUESTED** [review]

File: plugin/pkg/scheduler/scheduler.go (hunk 1 above)
Comment (bsalamat) [reply plugin/pkg/scheduler/scheduler.go]: Done.

**Comment by bsalamat** [comment]
@timothysc or @davidopp could one of you lgtm this? I think we should merge this ASAP.

**Review by timothysc — APPROVED** [review]
/lgtm

**Pull request merged**